fl4health/utils/dataset.py

```diff
@@ -39,36 +24,59 @@ def update_target_transform(self, g: Callable) -> None:
         else:
             self.target_transform = g
 
+    @abstractmethod
+    def __getitem__(self, index: int) -> Tuple[torch.Tensor, torch.Tensor]:
+        raise NotImplementedError
 
-class MnistDataset(BaseDataset):
-    def __init__(
-        self,
-        data_path: Path,
-        train: bool,
-        transform: Union[None, Callable] = None,
-        target_transform: Union[None, Callable] = None,
-    ):
-        mnist_dataset = MNIST(data_path, train=train, download=True)
-        self.data = mnist_dataset.data
-        self.targets = mnist_dataset.targets
-        self.transform = transform
-        self.target_transform = target_transform
+    @abstractmethod
+    def __len__(self) -> int:
+        raise NotImplementedError
 
 
-class Cifar10Dataset(BaseDataset):
+class TensorDataset(BaseDataset):
     def __init__(
         self,
-        data_path: Path,
-        train: bool,
-        transform: Union[None, Callable] = None,
-        target_transform: Union[None, Callable] = None,
-    ):
-        cifar10_dataset = CIFAR10(data_path, train=train, download=True)
-        self.data = torch.from_numpy(cifar10_dataset.data)
-        self.targets = torch.Tensor(cifar10_dataset.targets).long()
+        data: torch.Tensor,
+        targets: Optional[torch.Tensor] = None,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+    ) -> None:
+        super().__init__()
+        self.data: torch.Tensor = data
+        self.targets: Optional[torch.Tensor] = targets
         self.transform = transform
         self.target_transform = target_transform
 
+    def __getitem__(self, index: int) -> Tuple[torch.Tensor, torch.Tensor]:
+        assert self.targets is not None
+
+        data, target = self.data[index], self.targets[index]
+
+        if self.transform is not None:
```

Comment:
This assumes that any transforms that have been defined would initially operate on a numpy array, which I think is a little confusing, since data and target are torch.Tensors. I'd say maybe kill the numpy conversion and do it as part of any transforms that are set?